tests/global_communication/test.cpp

```diff
@@ -28,12 +38,44 @@ int main(int argc, char **argv) {
     // now add our custom printer
     listeners.Append(new mpi_listener("results_global_communication"));
 
-    // record the local return value for tests run on this mpi rank
-    //      0 : success
-    //      1 : failure
-    auto result = RUN_ALL_TESTS();
+    int return_value = 0;
+    try {
+        auto arg = argv+1;
+        while (*arg) {
+            if (auto comm_size = to::parse_opt<unsigned>(arg, 'd', "dryrun")) {
+                if (*comm_size==0) {
+                    throw to::parse_opt_error(*arg, "must be positive integer");
```

Comment:
`arg` is a `util::optional`, and the way that we take the underlying value inside an `optional` is via the `*` operator.

@halfflat, should the interface for `to::parse_opt_error()` also take an `optional` instead of the underlying value?